src/particles.cpp

```diff
@@ -131,12 +133,17 @@ void Particle::step(float dtime)
 		aabb3f box = m_collisionbox;
 		v3f p_pos = m_pos*BS;
 		v3f p_velocity = m_velocity*BS;
-		collisionMoveSimple(m_env, m_gamedef,
-			BS*0.5, box,
-			0, dtime,
-			&p_pos, &p_velocity, m_acceleration * BS);
-		m_pos = p_pos/BS;
-		m_velocity = p_velocity/BS;
+		collisionMoveResult r = collisionMoveSimple(m_env,
+				m_gamedef, BS*0.5, box, 0,
+				dtime, &p_pos, &p_velocity,
+				m_acceleration * BS);
+		if (m_collision_remove && r.collides) {
+			// force expiration of the particle
+			m_expiration = -1.0;
+		} else {
+			m_pos = p_pos/BS;
```

Comment:
can you add spaces between the p_pos and BS for better reading ? same thing on the previous lines